plugins/Compressor/CompressorControlDialog.cpp

```diff
@@ -0,0 +1,757 @@
+	connect(&m_controls->m_lookaheadModel, SIGNAL(dataChanged()), this, SLOT(lookaheadChanged()));
+	connect(&m_controls->m_limiterModel, SIGNAL(dataChanged()), this, SLOT(limiterChanged()));
+
+	m_timeElapsed.start();
+
+	peakmodeChanged();
+	stereoLinkChanged();
+	lookaheadChanged();
+	limiterChanged();
+}
+
+
+void CompressorControlDialog::makeLargeKnob(Knob * knob, QString hint, QString unit)
+{
+	knob->setHintText(hint, unit);
+	knob->setFixedSize(56, 56);
+	knob->setOuterRadius(23);
+	knob->setInnerRadius(15);
+	knob->setCenterPointX(28);
+	knob->setCenterPointY(28);
+}
+
+void CompressorControlDialog::makeSmallKnob(Knob * knob, QString hint, QString unit)
+{
+	knob->setHintText(hint, unit);
+	knob->setFixedSize(30, 30);
+	knob->setOuterRadius(10);
+	knob->setInnerRadius(4);
+	knob->setCenterPointX(15);
+	knob->setCenterPointY(15);
+}
+
+
+void CompressorControlDialog::peakmodeChanged()
+{
+	m_rmsKnob->setVisible(!m_controls->m_peakmodeModel.value());
+	m_rmsEnabledLabel->setVisible(!m_controls->m_peakmodeModel.value());
+}
+
+
+void CompressorControlDialog::stereoLinkChanged()
+{
+	m_blendKnob->setVisible(m_controls->m_stereoLinkModel.value() == 4);
+	m_blendEnabledLabel->setVisible(m_controls->m_stereoLinkModel.value() == 4);
+}
+
+
+void CompressorControlDialog::lookaheadChanged()
+{
+	m_lookaheadLengthKnob->setVisible(m_controls->m_lookaheadModel.value());
+	m_lookaheadEnabledLabel->setVisible(m_controls->m_lookaheadModel.value());
+}
+
+
+void CompressorControlDialog::limiterChanged()
+{
+	m_ratioKnob->setVisible(!m_controls->m_limiterModel.value());
+	m_ratioEnabledLabel->setVisible(!m_controls->m_limiterModel.value());
+}
+
+
+void CompressorControlDialog::updateDisplay()
+{
+	if (!isVisible())
+	{
+		m_timeElapsed.restart();
+		return;
+	}
+
+	int elapsedMil = m_timeElapsed.elapsed();
+	m_timeElapsed.restart();
+	m_timeSinceLastUpdate += elapsedMil;
+	int compPixelMovement = int(m_timeSinceLastUpdate / COMP_MILLI_PER_PIXEL);
+	m_timeSinceLastUpdate %= COMP_MILLI_PER_PIXEL;
+
+	// Time Change / Daylight Savings Time protection
+	if (!compPixelMovement || compPixelMovement <= 0)
+	{
+		return;
+	}
+
+	if (!m_controls->m_effect->isEnabled() || !m_controls->m_effect->isRunning())
+	{
+		m_controls->m_effect->m_displayPeak[0] = COMP_NOISE_FLOOR;
+		m_controls->m_effect->m_displayPeak[1] = COMP_NOISE_FLOOR;
+		m_controls->m_effect->m_displayGain[0] = 1;
+		m_controls->m_effect->m_displayGain[1] = 1;
+	}
+
+	const float peakAvg = (m_controls->m_effect->m_displayPeak[0] + m_controls->m_effect->m_displayPeak[1]) * 0.5f;
+	const float gainAvg = (m_controls->m_effect->m_displayGain[0] + m_controls->m_effect->m_displayGain[1]) * 0.5f;
+
+	QPainter p;
+	float yPoint = dbfsToYPoint(ampToDbfs(peakAvg));
+	float yGainPoint = dbfsToYPoint(ampToDbfs(gainAvg));
+
+	int threshYPoint = dbfsToYPoint(m_controls->m_effect->m_thresholdVal);
+	int threshXPoint = m_kneeWindowSizeY - threshYPoint;
+
+	p.begin(&m_visPixmap);
```

Comment:
I think this (and every new `p.begin()` "block") is a good candidate for a separate helper function, as to not make this `updateDisplay` function too complex. Then we just call the function `drawX(p)`.